packages/@ember/application/lib/application.js

```diff
@@ -516,7 +526,7 @@ const Application = Engine.extend({
     @method domReady
   */
   domReady() {
-    if (this.isDestroyed) {
+    if (this.isDestroying || this.isDestroyed) {
```

Comment:
FWIW, once we go `this.isDestroying = true` we never change it again. I think checking both makes the code easier to read (and avoids everyone having to internalize that particular detail) but I don't think it is actually going to change anything about the semantics.